### ISSUE TITLE
Generate import library without external tools

### ISSUE THREAD

**Issue by messense** [PyO3/python3-dll-a#20]
I've take some time to port `llvm-dlltool` to Rust, [implib-rs](https://github.com/messense/implib-rs) is a crate implements generating Windows import library from module definition file.

~~Unfortunately `rust-ar` [doesn't support writing symbol table](https://github.com/mdsteele/rust-ar/issues/16), so the currently it requires to run `ranlib` after generation to be useful. Thus it's not ready to be integrated in `python3-dll-a` right now. Hopefully someday we can find a solution to it.~~

I wonder how hard it is to port the mingw64 version, it'd be nice to support them both.

---

Update: [implib](https://github.com/messense/implib-rs) crate now produces the exact same `.lib` file like `llvm-dlltool` and it works for linking with `lld` with the [this `python3-dll-a` patch](https://github.com/PyO3/python3-dll-a/compare/implib):

<img width="1176" alt="image" src="https://user-images.githubusercontent.com/1556054/169633969-b7d6af69-9333-40b1-9683-e55a4739fdbf.png">

**Comment by ravenexp** [comment]
Please keep in mind that [RFC 2627](https://github.com/rust-lang/rust/issues/58713) is still being actively worked on, and when it hits stable it will make hacks like `xwin` or `python3-dll-a` obsolete.

**Comment by messense** [comment]
Looks like the implementation of RFC 2627 does not support `DATA`: https://github.com/rust-lang/rust/blob/10913c00018c76103b2fd4260d8c02ec728fd244/compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp#L1731 , will that be a problem for pyo3?

**Comment by ravenexp** [comment]
Yes, PyO3 uses quite a lot of `DATA` symbols for singletons like `_Py_NoneStruct`.

**Comment by messense** [comment]
And to support multiple Python versions and abi3 in `pyo3-ffi`, the `link` attributes are tedious with lots of `cfg`s.

```rust
#[link(name = "python3.dll", kind = "raw-dylib")]
#[link(name = "python37.dll", kind = "raw-dylib")]
#[link(name = "python38.dll", kind = "raw-dylib")]
#[link(name = "python39.dll", kind = "raw-dylib")]
#[link(name = "python310.dll", kind = "raw-dylib")]
#[link(name = "python311.dll", kind = "raw-dylib")]
```

And [this MINGW Python dll name thing](https://github.com/PyO3/python3-dll-a/pull/17#issuecomment-1123251092) is annoying too.

**Comment by ravenexp** [comment]
Yes, code like that is best generated by the build script or a macro.
